src/shared/ui/SearchForm/SearchForm.test.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { SearchForm } from "./SearchForm";
+import { keyboard } from "@testing-library/user-event/dist/keyboard";
+
+describe("SearchForm Component Functionality", () => {
+  test("Should Render Initial Value", () => {
+    const initialValue = "Test";
+    render(<SearchForm initialValue={initialValue} />);
+    const inputElement = screen.getByRole("textbox");
+    const inputValue = inputElement.value;
+    expect(inputValue).toBe(initialValue);
+  });
+  test("Should onChange fire with right value when clicked submit after typing", () => {
+    const mockOnSearch = jest.fn((e) => e.preventDefault());
+    const changedValue = "Javascript";
+
+    render(<SearchForm onSearch={mockOnSearch} />);
+
+    const input = screen.getByTestId("search-input");
+    const button = screen.getByTestId("search-button");
+
+    userEvent.type(input, changedValue);
+    userEvent.click(button);
+
+    // This part doesn't work, I didn't find a way to pass an event argument
+
+    // expect(mockOnSearch).toHaveBeenCalledWith(
+    //     expect.objectContaining({
+    //       target: {
+    //         value: {changedValue}
+    //       }
+    //     })
+    //   );
+
+    // Proper Value doesn't check
+    expect(mockOnSearch).toBeCalled();
+  });
+
+  test("Should onChange fire with right value when clicked enter after typing", () => {
+    const mockOnSearch = jest.fn((e) => e.preventDefault());
+    const changedValue = "Javascript";
```

Comment:
let's clarify that it's a mock data
```suggestion
    const mockedChangedValue = "Javascript";
```
considering it's a primitive value, in some projects you can face writing such names using uppercased approach `MOCKED_CHANGED_VALUE`

Reply:
done